pkg/agent/agent.go

```diff
@@ -417,12 +425,14 @@ func (c *agentController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	}
 
 	if !hc.GetDeletionTimestamp().IsZero() {
+		c.log.Info(fmt.Sprintf("hostedcluster %s has deletionTimestamp %s. Skip reconciling klusterlet secrets", hc.Name, hc.GetDeletionTimestamp()))
```

Comment:
not sure if `%s` should be used with `GetDeletionTimestamp()` which seems to be a `Time` pointer. Maybe ``GetDeletionTimestamp().String()` since we know it probably won't result in a nilpointer because the earlier if check.